src/app/services/tasksApi.ts

```diff
@@ -50,10 +49,13 @@ export const tasksApi = api.injectEndpoints({
             }),
         }),
         updateTask: builder.mutation<void, TaskRequestPayload>({
-            query: (task: TaskRequestPayload) => ({
-                url: `tasks/${task.id}`,
+            // isDevEnabled is the Feature flag for status update based on task status. This flag is temporary and will be removed once the feature becomes stable.
+            query: ({ task, id, isDevEnabled }: TaskRequestPayload) => ({
+                url: isDevEnabled
+                    ? `tasks/${id}?userStatusFlag=true`
+                    : `tasks/${id}`,
```

Comment:
```suggestion
                url: `tasks/${id}`,
                params: { userStatusFlag: isDevEnabled }
```

Reply:
`params: { userStatusFlag: isDevEnabled }` will always pass the query param flag even if its true or false, I want to call `tasks/${id}?userStatusFlag=true` only when the flag is true. if its false or undefined (since this is optional param) I don't wanna call the api with flag